### DISCUSSION TITLE
extensions don't work ?  what is wrong with my Dockfile

### DISCUSSION THREAD

**Discussion by c21xdx** [coder/code-server#4955] [Q&A]
```
FROM codercom/code-server

USER root

ENV GO_VERSION=1.17.8 \
    GOOS=linux \
    GOARCH=amd64 

RUN apt-get update && apt-get install -y cron curl  \
    && mkdir -p /home/coder/project/ \
    ## install golang
    && curl -fsSL https://go.dev/dl/go$GO_VERSION.$GOOS-$GOARCH.tar.gz | tar -C /usr/local -xzv \
    && echo "export PATH=$PATH:/usr/local/go/bin" > /etc/profile \
    ## install extensions
    && mkdir -p /home/coder/extensions && code-server --extensions-dir=/home/coder/extensions \
                                              --install-extension=golang.Go \
                                              --install-extension=MS-CEINTL.vscode-language-pack-zh-hans \
                                              --install-extension=Equinusocio.vsc-material-theme \
                                              --install-extension=PKief.material-icon-theme \
                                              --install-extension=Remisa.shellman \
                                              --install-extension=timonwong.shellcheck \
    && apt-get clean && rm -rf /var/lib/apt/lists/* /var/tmp/* /tmp/*

EXPOSE 8080

CMD exec code-server /home/coder/project --host 0.0.0.0
```

=======================================
all  extensions had download on /home/coder/extensions, but all are not work, what is wrong with me? thanks

![code](https://user-images.githubusercontent.com/5766149/156931505-55a2a24d-f188-4728-a1a6-7741f16d6fcc.jpg)

**Comment by lg-nineoneninethreenine** [comment]
Maybe it could be that you have to manually install some extensions. I think this does happen, so manual install is the best right now. If you want, you could use `wget` in your Dockerfile and manually install the VSIX files. It could be painstaking, but I think it could be easy since you will know beforehand what extensions they are and if you really need it.

If you want them to be installed on the Dockerfile itself, try `code-server --install-extension=extension-filename` on your Dockerfile. Use the VSIX as filename(tab autocompletion is your friend) from the Microsoft marketplace but remember, some of them might not work, like GitHub Copilot OOB, you have to do some tricks to fix it.

**Reply by lg-nineoneninethreenine** [reply]
If this is on GitHub, I could potentially help you out with this, if you are fine.

**Reply by lg-nineoneninethreenine** [reply]
You might think I am repeating some things, and yeah I did since I didn't read the description, I wanted to be quick. But I hope this solves it!

**Comment by jsjoeio** [comment]
Hmm...So in the screenshot it sounds like the extension have been installed, I think on this last line:
```dockerfile
CMD exec code-server /home/coder/project --host 0.0.0.0
```

You need to change it to:
```dockerfile
CMD exec code-server /home/coder/project --host 0.0.0.0 --extensions-dir=/home/coder/extensions
```

That way code-server knows where you're installed extensions are when it starts up.
